src/backend/interfaces/REST/py.server/caliopen_api/base/exception.py

```diff
@@ -34,22 +34,42 @@ class AuthenticationError(HTTPClientError):
 
     code: 401, title: Unauthorized
     """
+
+    def __init__(self, error=None):
+        if error:
```

Comment:
error may have no .message attribute and it's absolutely not a normal way to build an exception (give another exception on instanciation)

Reply:
I copied code from pyramid's httpexceptions package to build bespoke exception as subclass of HTTPClientError class.
Regarding the `message` property, it is a default property of python BaseException object, and we use this property to build the json error payload returned to client.

Reply:
If it's a subclass of HTTPClientError (HTTPException as parent) you must change the __init__ constructor only for correct reasons, hence you break the hierarchy and polymorphism of your classes, at least.

And error in your __init__ can be many things other than an exception, so error.message may not exist.

Something is not correct in the object design

Reply:
so, if I want to be able to handle an `error` passed as an argument when calling the HTTPException, how do I must code it ? Something like this ? : 

```
    def __init__(self, error=None, detail=None, headers=None, comment=None,
                 body_template=None, **kw):
        if error is not None and not insinstance(error, Exception):
            raise ValueError("HTTP Unprocessable need an error ")
        super(_HTTPClientError, self).__init__(
            detail=detail, headers=headers, comment=comment,
            body_template=body_template, error=error, **kw)
```

Reply:
Why do you want to handle error in the exception constructor ? Just handle it on callee when instanciate the exception.